### DISCUSSION TITLE
Light theme

### DISCUSSION THREAD

**Discussion by Sahraeidev** [TheHerobrine/DarkerNodes#146] [Q&A] [answered]
Hello i have a question when will you complete the light theme

**Comment by hugoattal** [comment, marked as answer]
Hello 🙂 ! I'm still working on it here: https://github.com/hugoattal/DarkerNodes/issues/17

Though, I must admit that it's insanely difficult, and I'm kinda stuck on it... There's a lot of part of the engine that are hard-coded and that I can modify using "standard" ways. So I must use hacks to patch the engine once it's loaded, but this has limited result. I've spent countless hours trying to make it work...

So it's kinda "in pause" until I find a way to do this. Anyone would simply say that it's impossible, but I still think there must be one way or another to do it. If you want to be kept posted on, I suggest you subscribe to the notification of the corresponding issue 😉

---

If you're interested in the "how" and have a bit of C++ knowledge, here is an example. Let say I want to change the color of this:
![image](https://user-images.githubusercontent.com/4563971/116746582-ff006200-a9fc-11eb-8b50-bca9e40700f7.png)

This is coded in a private method like this:
```cpp
FSlateColor SSceneOutliner::GetFilterStatusTextColor() const
{
	if ( !IsFilterActive() )
	{
		// White = no text filter
		return FLinearColor( 1.0f, 1.0f, 1.0f );
	}
	else if( FilteredActorCount == 0 )
	{
		// Red = no matching actors
		return FLinearColor( 1.0f, 0.4f, 0.4f );
	}
	else
	{
		// Green = found at least one match!
		return FLinearColor( 0.4f, 1.0f, 0.4f );
	}
}
```

As you can see, the colors are hard-coded here. I can't retrieve a pointer to that method since it's private, so not exported. Some functions that I would like to patch are inclined, some are too big (so instead of a "hook", I must use a "trampoline"). I'll try to find a way, but as I said, I'm kinda stuck for now...

**Reply by s-t-a-y** [reply]
How to update the Plugin from the Editor itself? Or do I need to keep updating from the launcher?

**Reply by hugoattal** [reply]
@s-t-a-y I'm sorry, but you must update the plugin from the launcher (closing the editor before), it's how Epic makes it work 😕
